src/main/java/usi/si/seart/config/SchedulerConfig.java

```diff
@@ -47,4 +61,19 @@ public void handleError(@NotNull Throwable t) {
             log.error("Unhandled exception occurred while performing a scheduled job.", t);
         }
     }
+
+
+
+    @Bean(name = "GitCloning")
+    public Executor asyncExecutor() {
+        ThreadPoolTaskExecutor executor = new ThreadPoolTaskExecutor();
+        executor.setCorePoolSize(0);
+        executor.setMaxPoolSize(maxPoolThreads);
+        executor.setQueueCapacity(10); // maximum number of tasks in the queue, after which more threads would be created
+        executor.setKeepAliveSeconds(60); // keep-alive time for idle threads
```

Comment:
To keep the code more legible, you might want to state your intentions as a JDoc rather than these inline comments.